common/web/keyboard-processor/src/keyboards/activeLayout.ts

```diff
@@ -333,9 +339,10 @@ namespace com.keyman.keyboards {
       // Save each percentage key width as a separate member (do *not* overwrite layout specified width!)
       var keyPercent: number, padPercent: number, totalPercent=0;
       for(let j=0; j<keys.length-1; j++) {
-        keyPercent=parseInt(keys[j]['width'],10)/totalWidth;
+        const key = keys[j] as ActiveKey; // already 'polyfilled' in prior loop
+        keyPercent = key.width/totalWidth;
         keys[j]['widthpc']=keyPercent;
```

Comment:
You've got `key.width` here and `keys[j]['widthpc']`. What is the reason for the inconsistent references? Is it possible to do `key.widthpc = keyPercent;` instead?

(Also, do we even need `keyPercent` and `padPercent` variables, given they are only ever getting the same values as `key.widthpc` and `key.padpc`?)

Reply:
The original code was written at a time when I was concerned about making support for "advanced-optimization" minification more difficult to eventually enable.  We've since gone ahead and stated that we've fully abandoned that idea, so there's no need to keep the latter pattern around.

Accordingly, I've simplified any references _I've edited in this PR_ to not use the `obj['key']` style of reference to known fields and properties.

Reply:
> (Also, do we even need keyPercent and padPercent variables, given they are only ever getting the same values as key.widthpc and key.padpc?)

I suppose I can eliminate them by making related code more compact, much like I did with (new) lines 362-363.